### DISCUSSION TITLE
Should the DSL objects types not include `undefined`?

### DISCUSSION THREAD

**Discussion by mikedidomizio** [danger/danger-js#1367] [General]
With regards to the DSL objects, on the [reference page](https://danger.systems/js/reference.html) and in the code it states:

> // The GitHub metadata. This covers things like PR info, comments and reviews on the PR, label metadata, commits with GitHub user identities and some useful utility functions for displaying links to files.
> 
> // Strictly speaking, `github` is a nullable type, if you are not using GitHub then it will be undefined. For the DSL convenience sake though, it is classed as non-nullable

The change was made in this [PR](https://github.com/danger/danger-js/pull/534) with this description:

>  I think a user of Danger will know whether they're using bitbucket or GitHub and will just not use the other one. So for users of danger I've made both types not null.

I agree, it's unlikely a user will use the wrong one.  I was wondering though for the sake of having accurate typings why it not remain `readonly github?: GitHubDSL`?  In CI the data will be populated if configured properly but for development it's helpful to be notified it's not necessarily guaranteed to have data.  The current typings are misleading which defeats the purpose of typings.

**Comment by orta** [comment]
I still agree with my original reasoning - so it's probably not going to change, it'll mean people either add a superfluous `if`+ `return` or use a bunch of `!` or `.?` for code which doesn't need it.
